firebase-vscode/webviews/SidebarApp.tsx

```diff
@@ -1,127 +1,169 @@
+import { MenuItem, PopupMenu } from "./components/ui/popup-menu/PopupMenu";
+import { ButtonGroup } from "./components/ui/ButtonGroup";
 
-export function SidebarApp() {
-  const env = useBroker("notifyEnv")?.env;
-  /**
-   * null - has not finished checking yet
-   * empty array - finished checking, no users logged in
-   * non-empty array - contains logged in users
-   */
-  const user = useBroker("notifyUserChanged")?.user;
-  const isLoadingUser = useBroker("notifyIsLoadingUser");
+const user = brokerSignal("notifyUserChanged", {
+  initialRequest: "getInitialData",
+});
+const isLoadingUser = brokerSignal("notifyIsLoadingUser");
+const project = brokerSignal("notifyProjectChanged");
+const env = brokerSignal("notifyEnv");
+const configs = brokerSignal("notifyFirebaseConfig", {
+  initialRequest: "getInitialData",
+});
+const hasFdcConfigs = brokerSignal("notifyHasFdcConfigs", {
+  initialRequest: "getInitialHasFdcConfigs",
+});
+const emulatorsRunningInfo = brokerSignal("notifyEmulatorStateChanged", {
+  initialRequest: "getEmulatorInfos",
+});
+const psqlString = brokerSignal("notifyPostgresStringChanged");
 
-  const configs = useBroker("notifyFirebaseConfig", {
-    initialRequest: "getInitialData",
+function Welcome() {
+  const configLabel = useComputed(() => {
+    return !hasFdcConfigs.value ? "dataconnect.yaml" : "firebase.json";
   });
-  const hasFdcConfigs =
-    useBroker("notifyHasFdcConfigs", {
-      initialRequest: "getInitialHasFdcConfigs",
-    }) ?? false;
 
-  const accountSection = (
-    <AccountSection
-      user={user!}
-      isMonospace={env?.isMonospace!}
-      isLoadingUser={isLoadingUser!}
-    />
+  return (
+    <>
+      <Body>
+        No <code>{configLabel}</code> detected in this project
+      </Body>
+      <Spacer size="medium" />
+      <VSCodeButton
+        onClick={() => {
+          broker.send("runFirebaseInit");
+        }}
+      >
+        Run firebase init
+      </VSCodeButton>
+    </>
   );
-  // Just render the account section loading view if it doesn't know user state
-  if (!user) {
-    return (
-      <>
-        <Spacer size="medium" />
-        Login to use the Firebase plugin
-        <Spacer size="small" />
-        {accountSection}
-      </>
-    );
-  }
-  if (!configs?.firebaseJson?.value || !hasFdcConfigs) {
-    const configLabel = !hasFdcConfigs ? "dataconnect.yaml" : "firebase.json";
-    return (
-      <>
-        {accountSection}
-        <p>
-          No <code>{configLabel}</code> detected in this project
-        </p>
-        <br />
-        <VSCodeButton
-          onClick={() => {
-            broker.send("runFirebaseInit");
-          }}
-        >
-          Run firebase init
-        </VSCodeButton>
-      </>
-    );
-  }
+}
 
-  return <SidebarContent configs={configs as any} />;
+function EmulatorsPanel() {
+  return emulatorsRunningInfo.value?.infos ? (
+    <Emulators emulatorInfo={emulatorsRunningInfo.value?.infos!} />
```

Comment:
Should be safe to simplify to:
```suggestion
    <Emulators emulatorInfo={emulatorsRunningInfo.value.infos} />
```